src/Bicep.LangServer/BicepCompilationManager.cs

```diff
@@ -24,15 +25,36 @@ public class BicepCompilationManager : ICompilationManager
         private readonly IWorkspace workspace;
         private readonly ILanguageServerFacade server;
         private readonly ICompilationProvider provider;
+        private readonly IModuleRestoreScheduler scheduler;
 
         // represents compilations of open bicep files
         private readonly ConcurrentDictionary<DocumentUri, CompilationContext> activeContexts = new ConcurrentDictionary<DocumentUri, CompilationContext>();
 
-        public BicepCompilationManager(ILanguageServerFacade server, ICompilationProvider provider, IWorkspace workspace)
+        public BicepCompilationManager(ILanguageServerFacade server, ICompilationProvider provider, IWorkspace workspace, IModuleRestoreScheduler scheduler)
         {
             this.server = server;
             this.provider = provider;
             this.workspace = workspace;
+            this.scheduler = scheduler;
+        }
+
+        public void RefreshCompilation(DocumentUri documentUri)
+        {
+            if(this.GetCompilation(documentUri) is null)
+            {
+                // the compilation we are refreshing no longer exists
+                return;
+            }
+
+            // TODO: This likely has a race condition if the user is modifying the file at the same time
```

Comment:
>likely [](http://example.com/codeflow?start=26&length=6)

FWIW, I haven't observed the race condition while typing in VS code with the OCI registry hooked up.